specification/search/data-plane/Azure.Search/preview/2025-08-01-preview/knowledgeagent.json

```diff
@@ -0,0 +1,529 @@
+        {
+          "$ref": "#/definitions/KnowledgeAgentMessageContent"
+        }
+      ],
+      "properties": {
+        "text": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "text"
+      ],
+      "description": "Text message type."
+    },
+    "KnowledgeAgentMessageImageContent": {
+      "type": "object",
+      "x-ms-discriminator-value": "image",
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeAgentMessageContent"
+        }
+      ],
+      "properties": {
+        "image": {
+          "type": "object",
+          "properties": {
+            "url": {
+              "type": "string",
+              "description": "The url of the image."
+            }
+          },
+          "required": [
+            "url"
+          ]
+        }
+      },
+      "required": [
+        "image"
+      ],
+      "description": "Text message type."
+    },
+    "KnowledgeAgentActivityRecord": {
+      "type": "object",
+      "discriminator": "type",
+      "properties": {
+        "id": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The ID of the activity record."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the activity record."
+        }
+      },
+      "required": [
+        "id",
+        "type"
+      ],
+      "description": "Base type for activity records."
+    },
+    "KnowledgeAgentSearchActivityRecord": {
+      "type": "object",
+      "x-ms-discriminator-value": "searchQuery",
+      "allOf": [
+        {
+          "$ref": "#/definitions/KnowledgeAgentActivityRecord"
+        }
+      ],
+      "properties": {
+        "targetIndex": {
+          "type": "string",
+          "description": "The target index for the retrieval activity."
+        },
+        "query": {
+          "type": "object",
+          "properties": {
+            "search": {
+              "type": "string",
+              "description": "The search string."
+            },
+            "filter": {
+              "type": "string",
+              "description": "The filter string."
+            }
+          },
+          "description": "The query details for the retrieval activity."
+        },
+        "queryTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The query time for this retrieval activity."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The count of documents retrieved."
+        },
+        "elapsedMs": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The elapsed time in milliseconds for the retrieval activity."
+        }
+      },
+      "description": "Represents a retrieval activity record."
+    },
+    "KnowledgeAgentModelQueryPlanningActivityRecord": {
+      "type": "object",
+      "x-ms-discriminator-value": "modelQueryPlanning",
```

Comment:
Same question as the other discriminator value change, bug in 2025-05-01-Preview or a deliberate change?

Reply:
deliberate